kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java

```diff
@@ -493,4 +495,55 @@ default void removeOwnerReference(HasMetadata owner) {
   default Optional<ObjectMeta> optionalMetadata() {
     return Optional.ofNullable(getMetadata());
   }
+
+  /**
+   * Initializes this {@link ObjectMeta} field with name and namespace (if this instance represents a namespaced resource)
+   * provided by the specified HasMetadata instance. This is a convenience method to avoid boilerplate, notably when using
+   * Server-Side Apply, when creating a new instance with only some fields of the original one. Calls
+   * {@link #setMetadata(ObjectMeta)} when done, if you want to further configure this instance's metadata, please use
+   * {@link #initMetadataBuilderNameAndNamespaceFrom(ObjectMeta)} instead, which <em>doesn't</em> sets the metadata, leaving it
+   * up to the user once configuration is finished.
+   *
+   * @param original a HasMetadata instance from which to retrieve the name and namespace
+   */
+  default void initNameAndNamespaceFrom(HasMetadata original) {
+    Objects.requireNonNull(original);
+    initNameAndNamespaceFrom(original.getMetadata());
+  }
+
+  /**
+   * Initializes this {@link ObjectMeta} field with name and namespace (if this instance represents a namespaced resource)
+   * provided by the specified ObjectMeta instance. This is a convenience method to avoid boilerplate, notably when using
+   * Server-Side Apply, when creating a new instance with only some fields of the original one. Calls
+   * {@link #setMetadata(ObjectMeta)} when done, if you want to further configure this instance's metadata, please use
+   * {@link #initMetadataBuilderNameAndNamespaceFrom(ObjectMeta)} instead, which <em>doesn't</em> sets the metadata, leaving it
+   * up to the user once configuration is finished.
+   *
+   * @param original an ObjectMeta instance from which to retrieve the name and namespace
+   */
+  default void initNameAndNamespaceFrom(ObjectMeta original) {
+    final ObjectMeta meta = initMetadataBuilderNameAndNamespaceFrom(original).build();
+    setMetadata(meta);
+  }
+
+  /**
+   * Creates and initializes a new {@link ObjectMetaBuilder} with name and namespace (if this instance represents a namespaced
+   * resource) provided by the specified ObjectMeta instance. This is a convenience method to avoid boilerplate, notably when
+   * using Server-Side Apply, when creating a new instance with only some fields of the original one. This method
+   * <em>doesn't</em> set the metadata on this object, since it assumes that further configuration will occur on the newly
+   * created ObjectMetaBuilder.
+   *
+   * @param original an ObjectMeta instance from which to retrieve the name and namespace
+   * @return a new ObjectMetaBuilder instance initialized with the name and namespace (if the resource it is called on is a
+   *         namespaced one) of the specified ObjectMeta
+   */
+  default ObjectMetaBuilder initMetadataBuilderNameAndNamespaceFrom(ObjectMeta original) {
```

Comment:
Let's make this one static - it does not need access to the current instance.

Reply:
Makes sense. I originally hoped that I could set the metadata on the current instance as well.

Reply:
```
static ObjectMetaBuilder initMetadataBuilderNameAndNamespaceFrom(ObjectMeta original)
```
gives you the usage:

```
new SomethingBuilder().withMetadata(HasMetadata.initMetadataBuilderNameAndNamespaceFrom(original)....build())...
```

An alternative which doesn't quite set the metadata, but allows you to do so more fluently:

```
static <A extends ObjectMetaFluent<A>> A withNameAndNamespace(A value, HasMetadata original) {
```

gives you the usage:

```
HasMetadata.withNameAndNamespace(new SomethingBuilder().withNewMetadata(), original)...endMetadata()...
```

Reply:
Actually, it does access the current instance to check if it is `Namespaced` or not to decide whether to use and set the namespace…

Reply:
@metacosm I think I was assuming based upon your other comment though it could instead be:

```
static ObjectMetaBuilder initMetadataBuilderNameAndNamespaceFrom(HasMetadata original)
```

But didn't make that change in the signature shown above.

Reply:
Gotcha. Sorry for being slow, it's been a long day. 😓